termwiz/src/escape/apc.rs

```diff
@@ -246,14 +246,134 @@ impl KittyImageData {
 
                 Ok(data)
             }
-            Self::SharedMem { .. } => {
-                log::error!("kitty image protocol via shared memory is not supported");
-                Err(std::io::ErrorKind::Unsupported.into())
-            }
+            Self::SharedMem {
+                name,
+                data_offset,
+                data_size,
+            } => read_shared_memory_data(&name, data_offset, data_size),
         }
     }
 }
 
+#[cfg(unix)]
+fn read_shared_memory_data(
+    name: &str,
+    data_offset: Option<u32>,
+    data_size: Option<u32>,
+) -> std::result::Result<std::vec::Vec<u8>, std::io::Error> {
+    use nix::sys::mman::{shm_open, shm_unlink};
+    use std::fs::File;
+    use std::os::unix::io::FromRawFd;
+
+    let raw_fd = shm_open(
+        name,
+        nix::fcntl::OFlag::O_RDONLY,
+        nix::sys::stat::Mode::empty(),
+    );
+    if raw_fd.is_err() {
+        log::warn!("Unable to open shared memory object {}", name);
+        return Err(std::io::Error::last_os_error());
+    };
+    let mut f = unsafe { File::from_raw_fd(raw_fd.unwrap()) };
```

Comment:
Generally speaking, it's best to avoid `.unwrap()` as it can lead to program termination if something happens that you didn't expect, and it doesn't explain why that is OK.

I'd suggest mapping the error value and using `?` to propagate it and avoid writing `.unwrap` here:

```suggestion
    let raw_fd = shm_open(
        name,
        nix::fcntl::OFlag::O_RDONLY,
        nix::sys::stat::Mode::empty(),
    ).map_err(|_| {
       let err = std::io::Error::last_os_error();
       std::io::Error::new(std::io::ErrorKind::Other, format!("shm_open {} failed: {:#}", name, err))
    })?;
    let mut f = unsafe { File::from_raw_fd(raw_fd) };
```

Reply:
Didn't know about `.map_err()` function, have applied your suggestion, thanks!